bip32/src/path.rs

```diff
@@ -167,6 +173,22 @@ impl From<&[u32]> for DerivationPath {
     }
 }
 
+impl TryFrom<u32> for DerivationPath {
+    type Error = Bip32Error;
+
+    fn try_from(v: u32) -> Result<Self, Self::Error> {
+        Ok(Self(vec![v]))
+    }
+}
+
+impl TryFrom<&str> for DerivationPath {
```

Comment:
should probably also `impl FromStr` here

Reply:
FromStr is actually already implemented for DerivationPath further down.  The TryFrom impl is really the same thing, but other parts of the code uses TryInto<DerivationPath> as generic parameters.

Reply:
I think the right way to go here is to just remove the `try_parse_path` function and implement it in the FromStr.  Then change all the tests to call `parse()` instead of `try_parse_path`.  TryFrom can likewise just delegated to `parse()`.  Let me know if that seems better.

Reply:
I vaguely recall trying this and it causing some sort of type related issue 🤔  but it would be better so if you want to try it, go for it

Reply:
gave it a shot.  Works.

Reply:
@prestwich got time to take another look and maybe merge/release?  Just poking in.  Not a huge rush